### PR TITLE
Allow overriding severity based on error code prefix match

### PR DESCRIPTION
**Pull request by jnooree**
See #195 for context.

### CODE REVIEW

**Comment by karthiknadig** [comment]
@jnooree Thanks for the PR. Could you add tests for this scenario?

you can find an example here:
https://github.com/microsoft/vscode-flake8/blob/6f638010c80d4ee17a4ad515c3666f6d7cea7796/src/test/python_tests/test_linting.py#L198-L258

File: bundled/tool/lsp_server.py [hunk 1]
```diff
@@ -126,7 +126,15 @@ def _get_severity(
...
-    value = severity.get(code, None) or severity.get(code_type, "Error")
+    value = None
+
+    while code and value is None:
+        value = severity.get(code, None)
+        code = code[:-1]
+
+    if value is None:
+        severity.get(code_type, "Error")
+
```
Comment (karthiknadig) [inline bundled/tool/lsp_server.py:137]: This implementation could be redone as:
```python
def _get_severity(code: str, code_type: str, severity: Dict[str, str]) -> lsp.DiagnosticSeverity:
    """Converts severity provided by linter to LSP specific value."""

    try:
        while code:
            if code in severity:
                return lsp.DiagnosticSeverity[severity[code]]
            code = code[:-1]

        if code_type in severity:
            return lsp.DiagnosticSeverity[severity[code_type]]
    except KeyError:
        pass

    return lsp.DiagnosticSeverity.Error
```

File: bundled/tool/lsp_server.py [hunk 2]
```diff
@@ -126,7 +126,15 @@ def _get_severity(
...
     """Converts severity provided by linter to LSP specific value."""
-    value = severity.get(code, None) or severity.get(code_type, "Error")
+    value = None
+
+    while code and value is None:
+        value = severity.get(code, None)
+        code = code[:-1]
+
+    if value is None:
+        severity.get(code_type, "Error")
```
Comment (karthiknadig) [inline bundled/tool/lsp_server.py:136]:
```suggestion
        value = severity.get(code_type, "Error")
```

**Review by karthiknadig — CHANGES_REQUESTED** [review]

File: bundled/tool/lsp_server.py (hunk 1 above)
Comment (jnooree) [reply bundled/tool/lsp_server.py:137]: This will require double lookup on the severity dictionary. If you don't mind that, I'll accept the changes.

Comment (karthiknadig) [reply bundled/tool/lsp_server.py]: Yeah I agree. Restored to what you had with the fix for `value` with `code_type`: 
```python
def _get_severity(code: str, code_type: str, severity: Dict[str, str]) -> lsp.DiagnosticSeverity:
    """Converts severity provided by linter to LSP specific value."""
    value = None
    while code and value is None:
        value = severity.get(code, None)
        code = code[:-1]

    if value is None:
        value = severity.get(code_type, "Error")

    try:
        return lsp.DiagnosticSeverity[value]
    except KeyError:
        pass

    return lsp.DiagnosticSeverity.Error
```

Comment (jnooree) [reply bundled/tool/lsp_server.py]: Great, thanks! The fix was applied in 8714960.

**Comment by jnooree** [comment]
Do the tests added in 670dccb suffice?

**Comment by karthiknadig** [comment]
> Do the tests added in [670dccb](https://github.com/microsoft/vscode-flake8/commit/670dccb69defe7884c83a001190597d5957ba6c3) suffice?

It looks good.

**Review by karthiknadig — APPROVED** [review]

**Review by anthonykim1 — APPROVED** [review]

**Pull request merged**